File: universal/src/main/java/com/github/fernthedev/universal/entity/GsonObject.java
package com.github.fernthedev.universal.entity;

import com.github.fernthedev.universal.GameObject;

import java.awt.*;

public class GsonObject extends GameObject {
    private float alpha = 1;

    private Color color;


    protected int health = 100;
    protected int coin;

    private int width,height;
    private float life;

    public int getHealth() {
        return health;
    }

    public int getCoin() {
        return coin;
    }

    public float getAlpha() {
        return alpha;
    }

    public Color getColor() {
        return color;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public float getLife() {
        return life;
    }

    @Override
    public void tick() {

    }

    @Override
    public void render(Graphics g) {

    }

    @Override
    public Rectangle getBounds() {
        return null;
    }
}

File: universal/src/main/java/com/github/fernthedev/universal/entity/GsonObject.kt
package com.github.fernthedev.universal.entity

import com.github.fernthedev.universal.GameObject

class GsonObject(go: GameObject) : GameObject(
    location = go.location.copy(),
    velX = go.velX,
    velY = go.velY,
    height = go.height,
    width = go.width,
    color = go.color,
    entityId = go.entityId,
    uniqueId = go.uniqueId
) {

    val alpha = 1f
    var health = 100
    var coin = 0
    var life = 0f

    override fun tick() {}
}